File: benchmark/door/door.go
package main

import (
	"fmt"
)

func main() {
	var door int = 1
	var incrementer = 0

	for current := 1; current <= 100000000; current++ {
		//fmt.Printf("Door %d ", current)

		if current == door {
			fmt.Printf("Door %d Open\n", current)
			incrementer++
			door += 2*incrementer + 1
		} else {
			//fmt.Printf("Closed\n")
		}
	}

}

File: benchmark/door/door.php
<?php
$start_time = microtime(true);
for ($i = 1; $i <= 100; $i++) {
	$root = sqrt($i);
	$state = ($root == ceil($root)) ? 'open' : 'closed';
	echo "Door {$i}: {$state}\n";
}
$end_time = microtime(true);
$duration = $end_time - $start_time;

echo "Process duration : " .  round($duration,4) . "sec.\n";
?>